src/platform/plugins/shared/discover/public/ebt_manager/scoped_discover_ebt_manager.ts

```diff
@@ -150,6 +180,91 @@ export class ScopedDiscoverEBTManager {
     });
   }
 
+  private async trackFieldUsageInQueryEvent({
+    eventName,
+    fieldNames,
+    fieldsMetadata,
+  }: {
+    eventName: FieldUsageInQueryEventName;
+    fieldNames: string[];
+    fieldsMetadata: FieldsMetadataPublicStart | undefined;
+  }) {
+    if (!this.reportEvent) {
+      return;
+    }
+
+    const isOnlyFullTextSearch = fieldNames.length === 1 && fieldNames[0] === FREE_TEXT;
+
+    const eventData: FieldUsageInQueryEventData = {
+      [FIELD_USAGE_EVENT_NAME]: eventName,
+    };
+
+    // Handle full text search vs field-specific queries
+    if (isOnlyFullTextSearch) {
+      eventData[FIELD_USAGE_IN_QUERY_FIELD_NAMES] = [FREE_TEXT];
+    } else if (fieldsMetadata) {
+      // Process actual field names
+      const fields = await this.getFieldsFromMetadata({
+        fieldsMetadata,
+        fieldNames,
+      });
+
+      // tracks ECS compliant fields with a field name and non-ECS compliant fields with a "<non-ecs>" label
+      const categorizedFields = fieldNames.map((fieldName) =>
+        fields[fieldName]?.short ? fieldName : NON_ECS_FIELD
+      );
+
+      eventData[FIELD_USAGE_IN_QUERY_FIELD_NAMES] = categorizedFields;
+    }
+
+    this.reportEvent(FIELD_USAGE_IN_QUERY_EVENT_TYPE, eventData);
+  }
+
+  public async trackSubmittingKQLQuery({
+    query,
+    fieldsMetadata,
+  }: {
+    query: Query;
+    fieldsMetadata: FieldsMetadataPublicStart | undefined;
+  }) {
+    if (typeof query.query !== 'string' || query.query === '') {
+      return;
+    }
+
+    const extractedFieldNames = [...new Set(getKqlFieldNamesFromExpression(query.query))];
+
+    // we discarded an empty query earlier, so if we're getting an empty array here it's a full text search
+    const fieldNames = extractedFieldNames.length === 0 ? [FREE_TEXT] : extractedFieldNames;
+
+    await this.trackFieldUsageInQueryEvent({
+      eventName: FieldUsageInQueryEventName.kqlQueryUpdate,
```

Comment:
Should we track Lucene separately?

Reply:
Ahh the thing is that parser seems to get correctly Lucene fields, so we do have them. The only missing piece would be adding another type for Lucene if we need it, so it's a low cost change, if my assumptions are correct

Reply:
Added here: [39a51ce](https://github.com/elastic/kibana/pull/232964/commits/39a51ce8b52fe902aad3082cbcb51d87ab581d9e)

I can revert if we don't want this event name

Reply:
I'm definitely ok with tracking Lucene query fields if we can, but I don't think we can rely on `getKqlFieldNamesFromExpression` for it. KQL and Lucene have some overlap so it works in some cases, but I tested locally and it will throw an exception for Lucene queries like `host: artifacts\.*`. This is one more area I'd ask @lukasolson about to see if we have a way to do it, and if we don't then we can just omit Lucene queries entirely.